src/csm_testing/tests/iuf_stages/__main__.py

```diff
@@ -0,0 +1,416 @@
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+This script runs IUF from process-media to prepare-images.
+"""
+
+import subprocess
+import os
+import sys
+import yaml
+import requests
+from requests.auth import HTTPBasicAuth
+import json
+from csm_testing.lib.iuf_common import get_nexus_credentials, get_ca_certificates, run_command
+from cray_product_catalog.query import ProductCatalog
+from csm_testing.lib.iuf_constants import MEDIA_DIR, CONFIGMAP_NAME, NAMESPACE, PRODUCT_NAME, PRODUCT_VERSION, NEXUS_URL
+
+REPO_URL = f"{NEXUS_URL}/service/rest/v1/repositories/yum/hosted"
+test_cases = 0
+
+# nexus_repo_setup, check_product_data, run_iuf_script 
+# are helper functions being used by stage operations
+    
+def nexus_repo_setup():
+    global test_cases
```

Comment:
This is unnecessary, since the variable isn't being assigned to in this function.

```suggestion
```